To: vim_dev@googlegroups.com
Subject: Patch 7.3.288
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.288
Problem:    has('python') may give an error message for not being able to load
	    the library after using python3.
Solution:   Only give the error when the verbose argument is true.
Files:	    src/if_python.c, src/if_python3.c


*** ../vim-7.3.287/src/if_python.c	2011-06-19 00:27:46.000000000 +0200
--- src/if_python.c	2011-08-21 17:03:52.000000000 +0200
***************
*** 368,374 ****
       * standard C extension libraries of one or both python versions. */
      if (python3_loaded())
      {
! 	EMSG(_("E836: This Vim cannot execute :python after using :py3"));
  	return FAIL;
      }
  #endif
--- 368,375 ----
       * standard C extension libraries of one or both python versions. */
      if (python3_loaded())
      {
! 	if (verbose)
! 	    EMSG(_("E836: This Vim cannot execute :python after using :py3"));
  	return FAIL;
      }
  #endif
*** ../vim-7.3.287/src/if_python3.c	2011-07-15 15:54:39.000000000 +0200
--- src/if_python3.c	2011-08-21 17:05:19.000000000 +0200
***************
*** 367,373 ****
       * standard C extension libraries of one or both python versions. */
      if (python_loaded())
      {
! 	EMSG(_("E837: This Vim cannot execute :py3 after using :python"));
  	return FAIL;
      }
  # endif
--- 367,374 ----
       * standard C extension libraries of one or both python versions. */
      if (python_loaded())
      {
! 	if (verbose)
! 	    EMSG(_("E837: This Vim cannot execute :py3 after using :python"));
  	return FAIL;
      }
  # endif
*** ../vim-7.3.287/src/version.c	2011-08-27 15:10:00.000000000 +0200
--- src/version.c	2011-08-28 15:59:11.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     288,
  /**/

-- 
FIRST SOLDIER:  So they wouldn't be able to bring a coconut back anyway.
SECOND SOLDIER: Wait a minute! Suppose two swallows carried it together?
FIRST SOLDIER:  No, they'd have to have it on a line.
                 "Monty Python and the Holy Grail" PYTHON (MONTY) PICTURES LTD

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
